### ISSUE TITLE
Bug using python3 and commissions

### ISSUE THREAD

**Issue by Mageti** [enigmampc/catalyst#147]
Dear Catalyst Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Python Version: `Python 3.5.2`
* How did you install Catalyst: `pip`
* Python packages: 
```
$ pip freeze | grep catalyst
enigma-catalyst==0.4.3
```

I run a modified version of the sample [rsi_profit_target.py](https://github.com/enigmampc/catalyst/blob/master/catalyst/examples/rsi_profit_target.py)
Everything works smoothly, but when I add `context.set_commission(maker=0.0015, taker=0.0025)` in the `initialize()` function, I have this beautiful stacktrace:
```
$ python rsi_profit_target.py [2018-01-09 13:21:40.963239] INFO: run_algo: running algo in paper-trading mode
[2018-01-09 13:21:41.893046] INFO: exchange_algorithm: initialized trading algorithm in backtest mode
[2018-01-09 13:21:41.972679] INFO: rsi: initializing algo
Traceback (most recent call last):
  File "rsi_profit_target.py", line 280, in <module>
    end=pd.to_datetime('2018-1-8', utc=True),
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/utils/run_algo.py", line 577, in run_algorithm
    stats_output=stats_output
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/utils/run_algo.py", line 358, in _run
    overwrite_sim_params=False,
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/exchange/exchange_algorithm.py", line 340, in run
    data, overwrite_sim_params
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/exchange/exchange_algorithm.py", line 298, in run
    data, overwrite_sim_params
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/algorithm.py", line 724, in run
    for perf in self.get_generator():
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/algorithm.py", line 622, in get_generator
    return self._create_generator(self.sim_params)
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/algorithm.py", line 593, in _create_generator
    self.initialize(*self.initialize_args, **self.initialize_kwargs)
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/algorithm.py", line 457, in initialize
    self._initialize(self, *args, **kwargs)
  File "rsi_profit_target.py", line 38, in initialize
    context.set_commission(maker=0.0015, taker=0.0025)
  File "/home/mageti/trading_test/venv/lib/python3.5/site-packages/catalyst/exchange/exchange_algorithm.py", line 133, in set_commission
    key = self.blotter.commission_models.keys()[0]
TypeError: 'dict_keys' object does not support indexing
```
Thanks in advance,
Mageti

**Comment by swilso793** [comment]
Clearly you're passing in d.keys() to your shuffle function. Probably this was written with python2.x (when d.keys() returned a list). With python3.x, d.keys() returns a dict_keys object which behaves a lot more like a set than a list. As such, it can't be indexed.

The solution is to pass list(d.keys()) (or simply list(d)) to shuffle.
Source: https://stackoverflow.com/questions/17322668/typeerror-dict-keys-object-does-not-support-indexing

**Comment by fredfortier** [comment]
Fixed in 0.4.4.
